alacritty_terminal/src/term/mod.rs

```diff
@@ -1017,6 +1064,15 @@ impl<T> Term<T> {
             text.push('\n');
         }
 
+        // If wide char is not part of the selection, but leading spacer is, include it
+        if line_end == self.grid.num_cols()
```

Comment:
It seems like `line_end` is zero based, but `self.grid.num_cols()`?

Reply:
If you look at where `line_end` is defined, you can see that it is not zero based. We check with `min(line_length, cols.end + 1)`, where `cols.end` is zero-based. So we add one because the final line length specifically isn't.

Reply:
and now if you look at the check above `line-end == Column(0)`, you'll see some odd logic.

Reply:
It's not actually that odd. You have to look at it more like a count, the count of included cells.

So even if it's 1-based, because the "first element" is at 1, that doesn't mean that a length of 0 doesn't exist. This will be the case when the entire line is empty for example.

Reply:
Could you name it `line_len`, since it's what you're trying to say to me.

Reply:
Sounds like a great idea, yeah.